cirq/circuits/circuit.py

```diff
@@ -1504,7 +1504,10 @@ def with_device(
         )
 
     def transform_qubits(
-        self, func: Callable[['cirq.Qid'], 'cirq.Qid'], *, new_device: 'cirq.Device' = None
+        self,
+        qubit_map: Union[Dict['cirq.Qid', 'cirq.Qid'], Callable[['cirq.Qid'], 'cirq.Qid']],
```

Comment:
this is technically a breaking change. Consider adding a keyword-only argument `func=None` that emits a deprecation warning and forwards to `qubit_map`

Reply:
thanks for pointing out. done.